src/test/java/org/mozilla/zest/test/v1/ZestAssignFromElementTest.java

```diff
@@ -0,0 +1,237 @@
+        ZestAssignFromElement fromElement = new ZestAssignFromElement("Var")
+                .whereElementIs("div")
+                .atIndex(1, true)
+                .selectContent();
+
+        // When
+        String result = assign(fromElement, testPageB);
+
+        // Then
+        assertEquals("<input type=\"submit\" value=\"Submit\" />", result);
+    }
+
+    @Test
+    public void filterByValueIsStartingWithChoiceAndSelectContentOfSecondItem() throws Exception {
+        // Given
+        ZestAssignFromElement fromElement = new ZestAssignFromElement("Var")
+                .whereAttributeIs("value", "Choice.*")
+                .atIndex(1)
+                .selectContent();
+
+        // When
+        String result = assign(fromElement, testPageB);
+
+        // Then
+        assertEquals("Choice 2", result);
+    }
+
+    @Test
+    public void selectActionAttributeOfSecondForm() throws Exception {
+        // Given
+        ZestAssignFromElement fromElement = new ZestAssignFromElement("Var")
+                .whereElementIs("form")
+                .atIndex(1)
+                .selectAttributeValue("action");
+
+        // When
+        String result = assign(fromElement, testPageB);
+
+        // Then
+        assertEquals("foo.html#site=456", result);
+    }
+
+    @Test
+    public void testSerialization() {
+        ZestAssignFromElement fromElement = new ZestAssignFromElement("Var")
+                .whereElementIs("form")
+                .atIndex(1)
+                .selectAttributeValue("action");
+
+        String str = ZestJSON.toString(fromElement);
+        ZestAssignFromElement fromElement2 = (ZestAssignFromElement) ZestJSON.fromString(str);
+
+        assertEquals(fromElement.getElementType(), fromElement2.getElementType());
+        assertEquals(fromElement.getVariableName(), fromElement2.getVariableName());
+        assertEquals(fromElement.getAttributeNameFilter(), fromElement2.getAttributeNameFilter());
+        assertEquals(fromElement.getAttributeValueFilter(), fromElement2.getAttributeValueFilter());
+        assertEquals(fromElement.getElementNameFilter(), fromElement2.getElementNameFilter());
+        assertEquals(fromElement.getReturnedAttributeName(), fromElement2.getReturnedAttributeName());
+        assertEquals(fromElement.getAttributeValueFilterFlags(), fromElement2.getAttributeValueFilterFlags());
+        assertEquals(fromElement.getElementIndex(), fromElement2.getElementIndex());
+        assertEquals(fromElement.areFilteredElementsReversed(), fromElement2.areFilteredElementsReversed());
+        assertEquals(fromElement.isReturningAttribute(), fromElement2.isReturningAttribute());
+        assertEquals(fromElement.isReturningElement(), fromElement2.isReturningElement());
+        assertEquals(fromElement.isFilteredByAttribute(), fromElement2.isFilteredByAttribute());
+        assertEquals(fromElement.isFilteredByElementName(), fromElement2.isFilteredByElementName());
+    }
+
+    @Test
```

Comment:
Could use `@Test(expected = ZestAssignFailException.class)` instead of using `ExpectedException`.